Spigot-Server-Patches/0420-mplement-Mob-Goal-API.patch

```diff
@@ -0,0 +1,662 @@
+From 8ff186ef2db1713c0ae519ebf5bcbe19b0eb4996 Mon Sep 17 00:00:00 2001
+From: MiniDigger <admin@minidigger.me>
+Date: Fri, 3 Jan 2020 16:26:19 +0100
+Subject: [PATCH] mplement Mob Goal API
+
+
+diff --git a/pom.xml b/pom.xml
+index 495e11f0f..4b3e36db0 100644
+--- a/pom.xml
++++ b/pom.xml
+@@ -119,6 +119,13 @@
+             <version>1.3</version>
+             <scope>test</scope>
+         </dependency>
++        <!-- for vanilla goal scanning -->
++        <dependency>
++            <groupId>io.github.classgraph</groupId>
++            <artifactId>classgraph</artifactId>
++            <version>4.8.47</version>
++            <scope>test</scope>
++        </dependency>
+     </dependencies>
+ 
+     <repositories>
+diff --git a/src/main/java/com/destroystokyo/paper/ai/MobGoalHelper.java b/src/main/java/com/destroystokyo/paper/ai/MobGoalHelper.java
+new file mode 100644
+index 000000000..bf4c7a37c
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/ai/MobGoalHelper.java
+@@ -0,0 +1,180 @@
++package com.destroystokyo.paper.ai;
++
++import com.google.common.collect.BiMap;
++import com.google.common.collect.HashBiMap;
++
++import net.minecraft.server.PathfinderGoal;
++
++import java.util.EnumSet;
++import java.util.HashSet;
++import java.util.Set;
++
++public class MobGoalHelper {
++
++    private static final BiMap<String, String> deobfuscationMap = HashBiMap.create();
++
++    static final Set<String> ignored = new HashSet<>();
++
++    static {
++        // TODO these kinda should be checked on each release, in case obfuscation changes
```

Comment:
not sure if dumping mojang names here is a license violation

Reply:
same goes for using them as our field names

Reply:
removed the comments

Reply:
removed the comments